orm/encoding/ormfield/uint64.go

```diff
@@ -47,3 +48,157 @@ func compareUint(v1, v2 protoreflect.Value) int {
 		return 1
 	}
 }
+
+// CompactUint64Codec encodes uint64 values using EncodeCompactUint64.
+type CompactUint64Codec struct{}
+
+func (c CompactUint64Codec) Decode(r Reader) (protoreflect.Value, error) {
+	x, err := DecodeCompactUint64(r)
+	return protoreflect.ValueOfUint64(x), err
+}
+
+func (c CompactUint64Codec) Encode(value protoreflect.Value, w io.Writer) error {
+	_, err := w.Write(EncodeCompactUint64(value.Uint()))
+	return err
+}
+
+func (c CompactUint64Codec) Compare(v1, v2 protoreflect.Value) int {
+	return compareUint(v1, v2)
+}
+
+func (c CompactUint64Codec) IsOrdered() bool {
+	return true
+}
+
+func (c CompactUint64Codec) FixedBufferSize() int {
+	return 9
+}
+
+func (c CompactUint64Codec) ComputeBufferSize(protoreflect.Value) (int, error) {
+	return c.FixedBufferSize(), nil
+}
+
+// EncodeCompactUint64 encodes uint64 values in 2,4,6 or 9 bytes.
```

Comment:
As you wrote, compact encoding is more suitable for small numbers (e.g. auto-incrementing ids). In which case, we can maybe save the maximum space for smaller ints, namely, for the 2 MSB:
- 0x00 -> 1 byte, for n<64
- 0x01 -> 2 bytes, for n<16384
- 0x10 -> 3 bytes, for n<2^22
- 0x11 -> 9 bytes for uint64 or 5 bytes for uint32

We'll have a the best space saving for integers up until 2^22 (4million), then it'll be worst than fixed length.

Reply:
I was thinking the 64 values in byte 1 is too small of an ID space to really worry about. But if we did take this approach I would push the third case to 4 or 5 bytes maybe

Reply:
Thanks for the estimations! I see your point of KBs of diff for small IDs vs GBs of diff for lots of them.

> In what cases do you think numbers <64 would be most used? ... those will likely number in the thousands and millions respectively over time.

Actually million is the order of magnitude I had in mind too for IDs on chain, which is why I proposed 1,2,3,9 (optimized for <4M).

But the million-to-billion range is where 1,2,3,9 really underperforms over other ones. And I think we should maybe consider that range to be plausible too. I'm good with 2,4,6,9 👍, seems like the best compromise overall.